frontend/lib/modules/lend_library/lend_library.dart

```diff
@@ -0,0 +1,335 @@
+import 'dart:convert';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:location/location.dart';
+import 'package:provider/provider.dart';
+
+import '../../app_scaffold.dart';
+import '../../common/socket_service.dart';
+import '../../common/form_input/input_fields.dart';
+import './lend_library_item_class.dart';
+import './lend_library_item_state.dart';
+import '../user_auth/current_user_state.dart';
+
+class LendLibrary extends StatefulWidget {
+  @override
+  _LendLibraryState createState() => _LendLibraryState();
+}
+
+class _LendLibraryState extends State<LendLibrary> {
+  List<String> _routeIds = [];
+  SocketService _socketService = SocketService();
+  InputFields _inputFields = InputFields();
+  Location _location = Location();
+
+  final _formKey = GlobalKey<FormState>();
+  var _filters = {
+    'title': '',
+    //'tags': '',
+    'lng': -999.0,
+    'lat': -999.0,
+  };
+  bool _loading = false;
+  String _message = '';
+  bool _canLoadMore = false;
+  int _lastPageNumber = 1;
+  int _itemsPerPage = 25;
+  bool _skipCurrentLocation = false;
+  bool _locationLoaded = false;
+
+  List<LendLibraryItemClass> _lendLibraryItems = [];
+  bool _firstLoadDone = false;
+  var _selectedLendLibraryItem = {
+    'id': '',
+  };
+
+  @override
+  void initState() {
+    super.initState();
+
+    _routeIds.add(_socketService.onRoute('getLendLibraryItems', callback: (String resString) {
+      var res = jsonDecode(resString);
+      var data = res['data'];
+      if (data['valid'] == 1) {
+        if (data.containsKey('lendLibraryItems')) {
+          _lendLibraryItems = [];
+          for (var lendLibraryItem in data['lendLibraryItems']) {
+            _lendLibraryItems.add(LendLibraryItemClass.fromJson(lendLibraryItem));
+          }
+          if (_lendLibraryItems.length == 0) {
+            _message = 'No results found.';
+          }
+        } else {
+          _message = 'Error.';
+        }
+      } else {
+        _message = data['msg'].length > 0 ? data['msg'] : 'Error.';
+      }
+      setState(() {
+        _loading = false;
+        _message = _message;
+        _lendLibraryItems = _lendLibraryItems;
```

Comment:
I might be wrong, I will test it out locally, but I don't think line 55 and these `variable = variable` lines are needed, since you've defined the `_lendLibraryItems ` variable outside of the function,  `_lendLibraryItems.add` is changing the value, and `setState` being called here should update the state of the variable. It doesn't matter if the variables are reassigned in the setState call or not, as long as it's called. But I will test first.

Reply:
Hmm, I could have sworn I've had issues with things not updating and needed to call setState.. But if it's working, want to open a new PR without them?

Reply:
https://api.flutter.dev/flutter/widgets/State/setState.html doesn't this mean that if we want to trigger a rebuild we must call setState?

Reply:
Yep, we still call `setState` but only remove lines where `variable = variable`. I will push an update

Reply:
This seems to say it is not good practice, even if just for readability. Functionally it seems you are correct though. https://stackoverflow.com/questions/52383575/empty-set-state-what-is-the-point
https://medium.com/@mehmetf_71205/setting-the-state-2809936fb79d

Reply:
Right, I just created this pr, let me know if it makes sense -> https://github.com/clover-coop/green_city_app/pull/9/files

Reply:
Approved!